src/FlowDiagnostics.jl

```diff
@@ -486,56 +486,89 @@ function (MLD::MixedLayerDepth)(i, j, k, grid, args...)
     return ifelse(kₘₗ == -1, -Inf, zₘₗ)
 end
 
+"""
+    $(SIGNATURES)
+
+An abstract mixed layer depth criterion where the mixed layer is defined to be
+`anomaly` + `pertubation` greater than the surface value of `anomaly`.
+
+$(SIGNATURES) types should provide a method for the function `anomaly` in the form
+`anomaly(criterion, i, j, k, grid, args...)`, and should have a property `pertubation`.
+"""
+abstract type AbstractAnomalyCriterion end
+
+@inline function (criterion::AbstractAnomalyCriterion)(i, j, k, grid, args...)
+    δ = criterion.pertubation
+
+    ref = (anomaly(criterion, i, j, grid.Nz, grid, args...) + anomaly(criterion, i, j, grid.Nz+1, grid, args...)) * convert(eltype(grid), 0.5)
+
+    val = anomaly(criterion, i, j, k, grid,args...)
+
+    return val < ref + δ
+end
+
+@inline function interpolate_from_nearest_cell(criterion::AbstractAnomalyCriterion, i, j, k, grid, args...)
+    δ = criterion.pertubation
+
+    ref = (anomaly(criterion, i, j, grid.Nz, grid, args...) + anomaly(criterion, i, j, grid.Nz + 1, grid, args...)) * convert(eltype(grid), 0.5)
+
+    k_val  = anomaly(criterion, i, j, k, grid, args...)
+    k⁺_val = anomaly(criterion, i, j, k + 1, grid, args...)
+
+    zₖ = znode(i, j, k, grid, Center(), Center(), Center())
+    z₊ = znode(i, j, k+1, grid, Center(), Center(), Center())
+
+    return zₖ + (z₊ - zₖ) * (ref + δ - k_val) / (k⁺_val - k_val)
+end
+
 """
     $(SIGNATURES)
 
 Defines the mixed layer to be the depth at which the density is more than `pertubation`
 greater than the surface density.
 
 When this model is used, the arguments `buoyancy` and `C` should be 
-supploed where `buoyancy` should be the buoyancy model, and `C` should be a named 
+supplied where `buoyancy` should be the buoyancy model, and `C` should be a named 
 tuple of `(; T, S)`, `(; T)` or `(; S)` (the latter two if the buoyancy model 
 specifies a constant salinity or temperature).
 """
-struct DensityAnomalyCriterion{FT}
+struct DensityAnomalyCriterion{FT} <: AbstractAnomalyCriterion
     pertubation :: FT
 end
```

Comment:
The density anomaly can be expressed in terms of a buoyancy anomaly, where you require additional information about 1) the gravitational acceleration used in the model and 2) the reference density. For example:

```julia
struct DensityAnomalyCriterion
    buoyancy_anomaly
    reference_density
    gravitational_acceleration
end
```

You can then build a constructor that uses some default values for the parameters.

This will help users understand how the parameters they are using affect their results leading to higher-quality science.

Reply:
I like this more